src/components/DownscaleModal.tsx

```diff
@@ -0,0 +1,79 @@
+import { useRef } from 'react'
+import { useClickAway } from 'react-use'
+import { RESIZE_LD } from '../adapters/config'
+import Button from './Button'
+import Modal from './Modal'
+
+interface DownscaleModalProps {
+  onClose: () => void
+  onDownscale: () => void
+  onUpgrade: () => void
+  image: HTMLImageElement
+}
+
+export default function DownscaleModal(props: DownscaleModalProps) {
+  const { onClose, onDownscale, onUpgrade, image } = props
+  const ref = useRef(null)
+
+  useClickAway(ref, onClose)
+
+  let { width, height } = image
+  const maxSize = RESIZE_LD
+  if (width > height) {
+    if (width > maxSize) {
+      height = Math.round(height * (maxSize / width))
+      width = maxSize
+    }
+  } else if (height > maxSize) {
+    width = Math.round(width * (maxSize / height))
+    height = maxSize
+  }
```

Comment:
This kind of mutation is not a good habit. No problem because it is a primitive value, but if we do it on object it can cause hard to debug bugs 😅 (I saw some bugs with this kind of usage in the editor 🤫)

Suggestion : 

```suggestion
  const { width, height } = useMemo(() => {
    const maxSize = RESIZE_LD

    if (image.width > image.height) {
      return {
        height: Math.round(image.height * (maxSize / image.width))
        width: maxSize
      }
    }
    return {
      width: Math.round(image.width * (maxSize / image.height))
      height: maxSize
    }
  }, [image.width, image.height])
```

Or create something like "clamp" helper.

Reply:
True, thanks!